core/embed/rust/src/ui/display/color.rs

```diff
@@ -1,12 +1,31 @@
 use crate::ui::lerp::Lerp;
 
+#[cfg(not(feature = "ui_colors32bit"))]
 #[derive(Copy, Clone, PartialEq, Eq)]
 pub struct Color(u16);
+#[cfg(feature = "ui_colors32bit")]
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub struct Color {
+    r: u8,
+    g: u8,
+    b: u8,
+}
 
 impl Color {
+    #[cfg(not(feature = "ui_colors32bit"))]
     pub const fn from_u16(val: u16) -> Self {
         Self(val)
     }
+
+    #[cfg(feature = "ui_colors32bit")]
+    pub const fn from_u16(val: u16) -> Self {
+        Self {
+            r: ((val >> 8) & 0xF8) as u8,
```

Comment:
The code works, but it lacks precision. The value 0xFFFF is converted to 0xF8FCF800. The lower bits should replicate the highest bits value. Please look at `gfx_color16_to_r/g/b()` functions in `gfx_color.h`.

Reply:
fixed https://github.com/trezor/trezor-firmware/commit/92df5bf66d3e6964ec166f0b3e8be1a7643f6c33